### DISCUSSION TITLE
How to ignore grammar in Tree-sitter?

### DISCUSSION THREAD

**Discussion by refaelsh** [nix-community/nixvim#2091] [Q&A] [answered]
I used to have this in my init.lua:
```lua
'nvim-treesitter/nvim-treesitter',
build = ':TSUpdate',
config = function()
  require("nvim-treesitter.configs").setup({
  ensure_installed = "all",
  ignore_install = { 'org' },
  auto_install = true,
  highlight = { enable = true },
  indent = { enable = true },
})
end
```
Now I have this in my configuration.nix:
```nix
treesitter = {
  enable = true;
  settings = {
    highlight.enable = true;
    indent.enable = true;
  };
};
```

My question so what do I do about `ignore_install = { 'org' },`? How to do in Nixvim?

Thanks in advance.

**Comment by khaneliman** [comment, marked as answer]
The default behavior of our nixvim module will install all grammars from nixpkgs. To do a similar config using treesitter managed grammars you'd do this:

```nix
treesitter = {
  enable = true;
  nixGrammars = false;

  settings = {
    auto_install = true;
    ensure_installed = "all";
    ignore_install = [ "org" ];
    highlight.enable = true;
    indent.enable = true;
  };
};
```

If you'd like to manage them all from nixpkgs using `nixGrammars` and `grammarPackages` in a similar way, you'd need to do something to set `grammarPackages` to a filtered list excluding the packages you want.

**Reply by refaelsh** [reply]
Yes. I understand. Thank you very much :-)
